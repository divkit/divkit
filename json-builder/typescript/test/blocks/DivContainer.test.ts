import {
    divCard,
    DivContainer,
    DivImage,
    DivSeparator,
    fixed,
    IDivData,
    reference,
    rewriteRefs,
    templateHelper,
} from '../../src';

describe('DivContainer tests', (): void => {
    it('should create card with templates', (): void => {
        const templates = {
            product: new DivContainer({
                items: [
                    new DivImage({
                        width: reference('top_image_width'),
                        height: fixed(40),
                        image_url: reference('top_image_url'),
                    }),
                    new DivSeparator({
                        height: reference('gap_height'),
                        delimiter_style: {
                            orientation: 'horizontal',
                            color: reference('gap_color'),
                        },
                    }),
                    new DivImage({
                        width: reference('bottom_image_width'),
                        height: fixed(40),
                        image_url: reference('bottom_image_url'),
                    }),
                ],
            }),
        };

        const divData: IDivData = {
            log_id: 'id',
            states: [
                {
                    state_id: 1,
                    div: templateHelper(templates).product({
                        top_image_url: 'https://image.url/top',
                        top_image_width: fixed(40),
                        bottom_image_url: 'https://image.url/bottom',
                        bottom_image_width: fixed(40),
                        gap_height: fixed(10),
                        gap_color: '#000',
                        paddings: {
                            left: 10,
                            right: 10,
                        },
                    }),
                },
            ],
        };

        expect(divCard(rewriteRefs(templates), divData)).toMatchSnapshot();
    });
});
