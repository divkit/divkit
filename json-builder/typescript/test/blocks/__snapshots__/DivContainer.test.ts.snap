// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DivContainer tests should create card with templates 1`] = `
{
  "templates": {
    "product": {
      "type": "container",
      "items": [
        {
          "type": "image",
          "height": {
            "type": "fixed",
            "value": 40
          },
          "$image_url": "top_image_url",
          "$width": "top_image_width"
        },
        {
          "type": "separator",
          "delimiter_style": {
            "orientation": "horizontal",
            "$color": "gap_color"
          },
          "$height": "gap_height"
        },
        {
          "type": "image",
          "height": {
            "type": "fixed",
            "value": 40
          },
          "$image_url": "bottom_image_url",
          "$width": "bottom_image_width"
        }
      ]
    }
  },
  "card": {
    "log_id": "id",
    "states": [
      {
        "state_id": 1,
        "div": {
          "type": "product",
          "top_image_url": "https://image.url/top",
          "top_image_width": {
            "type": "fixed",
            "value": 40
          },
          "bottom_image_url": "https://image.url/bottom",
          "bottom_image_width": {
            "type": "fixed",
            "value": 40
          },
          "gap_height": {
            "type": "fixed",
            "value": 10
          },
          "gap_color": "#000",
          "paddings": {
            "left": 10,
            "right": 10
          }
        }
      }
    ]
  }
}
`;
